include/mbgl/gfx/vertex_attribute.hpp

```diff
@@ -389,7 +389,7 @@ class VertexAttributeArray {
 
 protected:
     template <typename DataDrivenPaintProperty, typename Evaluated>
-    static bool isConstant(const Evaluated& evaluated) {
+    static bool isConstant(const Evaluated& evaluated) noexcept {
```

Comment:
Agreed, any compile-time checking we can do would be great.

This is a relatively easy case because `DataDrivenPaintProperty` isn't really an arbitrary type, it's always a `PossiblyEvaluatedPropertyValue<T>` implementations of which have a simple `isConstant` that doesn't use `T`, just checks which variant alternative is present.

Reply:
I got it to work locally, but it needs more work for other platforms, apparently.